sklearn/ensemble/weight_boosting.py

```diff
@@ -147,7 +147,12 @@ def fit(self, X, y, sample_weight=None):
             # Early termination
             if sample_weight is None:
                 break
-
+            if not np.isfinite(np.sum(sample_weight)):
```

Comment:
Can we move this check a few lines down till:

```py
sample_weight_sum = np.sum(sample_weight)
````

and use that sum to do the check:

```py
if not np.isfinite(sample_weight_sum):
```

so we do not need to recompute the sum.